gateway/ln-gateway/src/lib.rs

```diff
@@ -1313,6 +1313,27 @@ impl Gateway {
         Ok(channels)
     }
 
+    /// Withdraws funds from the gateway's lightning node on-chain wallet.
+    pub async fn handle_withdraw_onchain_msg(
+        &self,
+        WithdrawOnchainPayload {
+            address,
+            amount: amount_sats,
+            fee_rate_sats_per_vbyte,
+        }: WithdrawOnchainPayload,
+    ) -> Result<Txid> {
+        let context = self.get_lightning_context().await?;
+        let response = context
+            .lnrpc
+            .withdraw_onchain(
+                address.assume_checked(),
+                amount_sats,
+                fee_rate_sats_per_vbyte,
+            )
+            .await?;
+        Ok(Txid::from_str(&response.txid).unwrap())
```

Comment:
Let's use `expect`

Reply:
Decided to return an actually `Err` instead

Reply:
Looks like `unwrap` is still used though?

Reply:
Sorry, just pushed